src/fu-engine.c

```diff
@@ -1506,33 +1506,35 @@ fu_engine_install_blob (FuEngine *self,
 	}
 
 	/* do the update */
-	if (!fu_plugin_runner_update_detach (plugin, device, &error_local)) {
-		fu_device_set_update_state (device, FWUPD_UPDATE_STATE_FAILED);
-		fu_device_set_update_error (device, error_local->message);
-		if ((flags & FWUPD_INSTALL_FLAG_NO_HISTORY) == 0 &&
-		    !fu_history_modify_device (self->history, device,
-					       FU_HISTORY_FLAGS_MATCH_OLD_VERSION,
-					       error)) {
+	if (!fu_device_has_flag (device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER)) {
```

Comment:
This commit makes sense to me; but while you're changing it I wonder if it makes sense to split some of it into a helper function.  A large chunk of the code below is duplicated 3 times in this function to get the new device.